src/GCodes/GCodes2.cpp

```diff
@@ -2381,6 +2381,98 @@ bool GCodes::HandleMcode(GCodeBuffer& gb, const StringRef& reply)
 		result = SetHeaterModel(gb, reply);
 		break;
 
+	case 308: // Set generic heater state
+	{
+		/*
+		 * M308 Generic Heater Control
+		 * 	Pnnn Heater number
+		 * 	Snnn Active Temp
+		 * 	Rnnn Standby Temp
+		 * 	Tnnn State: 0 = off, 1 = standby 2 = on.
+		 */
+		Heat& heat = reprap.GetHeat();
+		bool seen = false;
+		uint8_t currentHeater = 0;
+
+		if (gb.Seen('P')) {
+			currentHeater = gb.GetUIValue();
+			if (currentHeater < 0 || currentHeater >= NumHeaters)
```

Comment:
```suggestion
			if (currentHeater >= NumHeaters)
```
Since `currentHeater` is of type `uint8_t` it cannot be `< 0`.